user/test/com/google/gwt/user/rebind/rpc/testcases/client/SimplifiedRecursiveTypeGraphInstantiability.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2025 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.gwt.user.rebind.rpc.testcases.client;
+
+import com.google.gwt.user.client.rpc.IsSerializable;
+
+public interface SimplifiedRecursiveTypeGraphInstantiability extends IsSerializable {
+  /**
+   * Not serializable; interface only.
+   */
+  interface A extends IsSerializable {
+  }
+  
+  /**
+   * Default serializable, but with back-reference to B for which the question of instantiable
+   * subtypes depends on the serializability of this class. The reference to {@link C} which has a
+   * back-reference to {@link A} and B being its only subtypes candidate helps reproduce issue
+   * 10181.
+   */
+  class B implements A {
+    B b;
+    C c;
+  }
```

Comment:
Tests should permutate the order of these  (fields here and in the non-simple vers, the interfaces implemented, the root type requested, etc) - if we later introduce/discover a case where we fail only for a particular encounter order, we'll have to fix it again. 

Probably best to inline these classes as strings in the test, make java resources from them as other tests do, and repeat the test with different orderings as needed.

Likewise, enforcing stability in the traversal order of types (fields already have source order enforced if i'm not mistaken) will prevent the inconsistent nature of the bug, as discussed in the ticket.

Reply:
> Con: it conincidentally does not fail, and another problem in the algorithm stays looming for a potentially longer time. But I guess this is not on me to decide.

I think that's fair, but I don't think stochastic testing in downstream applications is the way to find that - or at least, it should be opt-in. I'd be comfortable with the sorting being optional, so it is consistent by default, but if someone wants to fuzz test (or we want in-build fuzz tests) we can support that.